google/cloud/spanner/value.cc

```diff
@@ -88,6 +88,83 @@ bool Equal(google::spanner::v1::Type const& pt1,
   }
 }
 
+std::ostream& StreamHelper(std::ostream& os, google::protobuf::Value const& v,
+                           google::spanner::v1::Type const& t) {
+  if (v.kind_case() == google::protobuf::Value::kNullValue &&
+      t.code() != google::spanner::v1::ARRAY &&
+      t.code() != google::spanner::v1::STRUCT) {
+    return os << "";
+  }
+
+  int i = 0;
```

Comment:
I think the observation was that the `i` could be local to the `for` loops.  Its use after the loop could be replaced with another expression given that the loops don't `break`.  But it also looks the the "size - 1" loops could be replaced with full loops by using the `std::string sep` idiom.

Reply:
Right. For example, I think the array case might look like this:

```cc
    case google::spanner::v1::ARRAY: {
      os << '[';
      char const* sep = nullptr;
      for (auto const& val : v.list_value().values()) {
        if (sep != nullptr) os << sep;
        StreamHelper(os, val, t.array_element_type());
        sep = ", ";
      }
      return os << ']';
    }
```

Struct could simplify to something like this as well.

Reply:
Done.